### PR TITLE
Refactor for 550 :  Introduction of a class for handling Personium URL 

### PR DESCRIPTION
**Pull request by shimono**
## Variety of Personium URL's
Personium unit needs to handle various URL's such as:
- Unit URL
- Cell URL 
- Role URL
- Box URL 
- etc...

Also personium now uses the following custom scheme
- personium-localunit
- personium-localcell
- personium-localbox

Overmore, it now runs with the following two configuration modes.:
- Subdomain-based cell url
- Path-based cell url

In current implementation of personium-core,  interpretation, conversion, normalization, and checking of these varieties of URL's are handled in various classes in various layers, which results in the complexity of the code.

# Introduction of PersoniumUrl class

By introducing a new class PersoniumUrl, which handles all the interpretation, conversion, normalization, and checking against URL's handled in Personium, the whole code of personium-core will be more neat, organized and easy-to-maintain.

### CODE REVIEW

File: src/main/java/io/personium/core/utils/PersoniumUrl.java [hunk 1]
```diff
@@ -0,0 +1,695 @@
...
+    /**
+     * enum indicating the Scheme type of the URL.
+     */
+    public static enum SchemeType {
+        /** enum value indicating "http" or "https" scheme. */
+        HTTP,
+        /** enum value indicating "ws" or "wss" scheme. */
+        WS,
+        /**
+         * enum indicating "personium-localunit" scheme using a single colon.
```
Comment (tochi-y) [inline src/main/java/io/personium/core/utils/PersoniumUrl.java]: Please add examples.

Comment (shimono) [reply src/main/java/io/personium/core/utils/PersoniumUrl.java]: example added. Thank you.

File: src/main/java/io/personium/core/utils/PersoniumUrl.java [hunk 2]
```diff
@@ -0,0 +1,696 @@
...
+        }
+        return sb.toString();
+    }
+
+    /**
+     *
+     * @deprecated
+     * @return
+     */
+    public String normalize() {
```
Comment (shimono) [inline src/main/java/io/personium/core/utils/PersoniumUrl.java]: This method will be removed in a following PR.

File: src/main/java/io/personium/core/utils/PersoniumUrl.java [hunk 3]
```diff
@@ -0,0 +1,696 @@
...
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Try to convert to a URL string with "personium-localunit" scheme.
+     * returns original url when the resourceType is EXTERNAL_UNIT or INVALID.
+     * @return an URL string representation using "personium-localunit" scheme.
+     */
+    public String toLocalcell() {
```
Comment (shimono) [inline src/main/java/io/personium/core/utils/PersoniumUrl.java]: This method is broken but not used yet.
It is fixed in a following PR.

**Pull request merged**